### PR TITLE
[ASTPrinter] Restore printing generic requirement contexts

### PR DESCRIPTION
**Pull request by bitjammer**
cfe9e6a3defb0368c2ef9742b59bf1d83ca66ecf removed calls to pre/post
printing of PrintStructureKind::GenericRequirement, so SourceKit DocInfo
requests started dropping the markers for generic requirements, causing
some weirdness with documentation rendering and post-processing.

Restore the calls to printStructPre/Post when printing generic
requirements.

rdar://problem/30561880

### CODE REVIEW

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by akyrtzi** [comment]
LGTM!

**Comment by bitjammer** [comment]
@swift-ci Please test

File: test/SourceKit/CursorInfo/cursor_info.swift [hunk 1]
```diff
@@ -422,31 +422,29 @@ func convention7(_: @convention(witness_method) ()->()) {}
...
+// CHECK33: <Declaration>struct S2&lt;T, U&gt; where T == U</Declaration>
+// CHECK33-NEXT: <decl.struct><syntaxtype.keyword>struct</syntaxtype.keyword> <decl.name>S2</decl.name>&lt;<decl.generic_type_param usr="s:tV11cursor_info2S21TMx"><decl.generic_type_param.name>T</decl.generic_type_param.name></decl.generic_type_param>, <decl.generic_type_param usr="s:tV11cursor_info2S21UMq_"><decl.generic_type_param.name>U</decl.generic_type_param.name></decl.generic_type_param>&gt; <syntaxtype.keyword>where</syntaxtype.keyword> <decl.generic_type_requirement>T == U</decl.generic_type_requirement></decl.struct>
 
 // RUN: %sourcekitd-test -req=cursor -pos=81:8 %s -- -F %S/../Inputs/libIDE-mock-sdk -I %t.tmp %mcp_opt %s | %FileCheck %s -check-prefix=CHECK34
 // CHECK34: source.lang.swift.decl.function.method.instance (81:8-81:50)
 // CHECK34-NEXT: foo(_:)
-// CHECK34-NEXT: s:FV11cursor_info2S23foou0_Rd_0_zqd__rFFT_T_FT_T_
-// CHECK34-NEXT: <T, U, V, W where U == T, W == V> (S2<T, U>) -> (() -> ()) -> () -> ()
```
Comment (benlangmuir) [inline test/SourceKit/CursorInfo/cursor_info.swift]: Did you mean to delete these other lines?  Also in other changes below.

Comment (bitjammer) [inline test/SourceKit/CursorInfo/cursor_info.swift]: Nope, I sure didn't. Sigh.

Comment (bitjammer) [inline test/SourceKit/CursorInfo/cursor_info.swift]: Ah, this was a conflict resolution when I cherry-picked the original commit. I'll fix it.

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by benlangmuir** [comment]
Thanks, LGTM!

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by bitjammer** [comment]
@shahmishal, I feel like the bot shouldn't be posting notifications that say "Build failed" for automatically restarted builds due to a different SHA.

**Comment by bitjammer** [comment]
(Especially since the build actually successfully starts)

**Pull request merged**